File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_grid.c
#include "ses_defines.h"
#include "ses_globals.h"
#include "ses_internals.h"
#include "ses_externs.h"

#include "stdio.h"
#include <stdlib.h>

#include <stdio.h>

#undef DEBUG_WRAP

#ifdef LC_UNDER
ses_boolean ses_get_grid_(ses_file_handle* pt_handle, ses_material_id* the_mid, ses_table_id* the_tid, long* nr, long* nt, long* ntab) {
#else

#ifdef UC_UNDER
ses_boolean SES_GET_GRID_(ses_file_handle* pt_handle, ses_material_id the_mid, ses_table_id the_tid, long* nr, long* nt, long* ntab) {
#else

#ifdef LC_NOUNDER
ses_boolean ses_get_grid(ses_file_handle* pt_handle, ses_material_id the_mid, ses_table_id the_tid, long* nr, long* nt, long* ntab) {
#else

#ifdef UC_NOUNDER
ses_boolean SES_GET_GRID(ses_file_handle* pt_handle, ses_material_id the_mid, ses_table_id the_tid, long* nr, long* nt, long* ntab) {
#endif

#endif
#endif
#endif

  ses_file_handle the_handle = *pt_handle;
  ses_material_id my_mid = *the_mid;
  ses_table_id my_tid = *the_tid;

#ifdef DEBUG_WRAP
  printf("wrap_ses_get_grid.c:  the_mid is %d the_tid is %d\n", my_mid, my_tid);
  printf("wrap_ses_get_grid.c:  calling ses_get_grid *nr is %ld *nt is %ld\n", *nr, *nt);
#endif

  ses_boolean return_value = ses_get_grid(the_handle, my_mid, my_tid, nr, nt, ntab);

  return return_value;

}

File: Source/ses_io/src/wrappers/fortran/wrap_ses_get_grid.f90

#undef DEBUG_WRAP

SES_BOOLEAN function ses_get_grid_f90(the_handle, the_mid, the_tid, arg_nr, arg_nt, arg_ntab) 

   implicit none

   SES_FILE_HANDLE, INTENT(IN) :: the_handle
   SES_MATERIAL_ID, INTENT(IN) :: the_mid
   SES_TABLE_ID, INTENT(IN) :: the_tid
   LONG, INTENT(OUT) :: arg_nr
   LONG, INTENT(OUT) :: arg_nt
   LONG, INTENT(OUT) :: arg_ntab

   SES_BOOLEAN :: return_value, ses_get_grid

   LONG, TARGET :: nr, nt, ntab
   LONG_POINTER :: pt_nr, pt_nt, pt_ntab
   pt_nr => nr
   pt_nt => nt
   pt_ntab => ntab

   nr = 0
   nt = 0
   ntab = 0

#ifdef DEBUG_WRAP
   print *, "SES_GET_GRID_F90: the_mid is ", the_mid
   print *, "SES_GET_GRID_F90: the_tid is ", the_tid
   print *, "SES_GET_GRID_F90: nr is ", nr
   print *, "SES_GET_GRID_F90: nt is ", nt
#endif

   return_value = ses_get_grid(the_handle, the_mid, the_tid, pt_nr, pt_nt, pt_ntab)

   arg_nr = nr
   arg_nt = nt
   arg_ntab = ntab

   ses_get_grid_f90 = return_value

end function ses_get_grid_f90

